server.js

```diff
@@ -168,69 +224,42 @@ if (cluster.isMaster) {
     });
 
     app.get('/proxy/*', function(req, res, next) {
-        // look for request like http://localhost:8080/proxy/http://example.com/file?query=1
-        var remoteUrl = getRemoteUrlFromParam(req);
-        if (!remoteUrl) {
-            // look for request like http://localhost:8080/proxy/?http%3A%2F%2Fexample.com%2Ffile%3Fquery%3D1
-            remoteUrl = Object.keys(req.query)[0];
-            if (remoteUrl) {
-                remoteUrl = url.parse(remoteUrl);
-            }
-        }
-
-        if (!remoteUrl) {
-            return res.status(400).send('No url specified.');
-        }
-
-        if (!remoteUrl.protocol) {
-            remoteUrl.protocol = 'http:';
-        }
-
-        var proxy;
-        if (upstreamProxy && !(remoteUrl.host in bypassUpstreamProxyHosts)) {
-            proxy = upstreamProxy;
-        }
-
-        //If you want to run a CORS proxy to data source, remove this section
-        if (!proxyAllowedHost(remoteUrl.host)) {
-            res.status(400).send('Host is not in list of allowed hosts: ' + remoteUrl.host);
-            return;
-        }
-
-        // encoding : null means "body" passed to the callback will be raw bytes
+        doProxy(req, res, next, function(remoteUrl, filteredRequestHeaders, proxy) {
+            return request.get({
+                url : url.format(remoteUrl),
+                headers : filteredRequestHeaders,
+                encoding : null,
+                proxy : proxy
+            }, function(error, response, body) {
+                var code = 500;
+
+                if (response) {
+                    code = response.statusCode;
+                    res.header(filterHeaders(req, response.headers));
+                }
 
-        var proxiedRequest;
-        req.on('close', function() {
-            if (proxiedRequest) {
-                proxiedRequest.abort();
-            }
+                res.status(code).send(body);
+            });
         });
+    });
 
-        var filteredReqHeaders = filterHeaders(req, req.headers);
-        if (!filteredReqHeaders['x-forwarded-for']) {
-            filteredReqHeaders['x-forwarded-for'] = req.connection.remoteAddress;
-        }
-
-        // http basic auth
-        var authRequired = proxyAuth[remoteUrl.host];
-        if (authRequired) {
-            filteredReqHeaders['authorization'] = authRequired.authorization;
-        }
-
-        proxiedRequest = request.get({
-            url : url.format(remoteUrl),
-            headers : filteredReqHeaders,
-            encoding : null,
-            proxy : proxy
-        }, function(error, response, body) {
-            var code = 500;
-
-            if (response) {
-                code = response.statusCode;
-                res.header(filterHeaders(req, response.headers));
-            }
+    app.post('/proxy/*', function(req, res, next) {
```

Comment:
Yep, should be fine.  The proxy service is always at the root relative to the node web server.  The reverse proxy, if any, rewrites URLs so it appears to be elsewhere.